packages/simple-validators/src/validators.ts

```diff
@@ -207,22 +206,122 @@ export function validateEnum<T>(
   return input;
 }
 
+// This pattern implements the date pattern from ISO8601, with a trailing capture group for timestamps
+const ISO8601_DATE_PATTERN = /^(\d\d\d\d)(?:-(\d\d))?(?:-(\d\d))?(T.*)?$/;
```

Comment:
This looks like it could also parse just a YEAR or YYYY-MM? Is that true - what happens then?

`(new Date('2023')).getUTCFullYear()`

Does this behave correctly? 🧪!

Reply:
`YYYY` is a valid ISO8601 date, but actually I don't think we want to support it now that I come back to this PR. There's a profile of ISO8601, RFC3339, that I think is better describing what we _do_ want: 
`YYYY-MM-DD`. I'll make this stricter.